sys/mips/gt64120.c

```diff
@@ -309,15 +295,51 @@ static int gt_pci_attach(device_t *pcib) {
   bus_write_1(io, PIIX_REG_ELCR + 0, LO(gtpci->elcr));
   bus_write_1(io, PIIX_REG_ELCR + 1, HI(gtpci->elcr));
 
-  pci_bus_enumerate(pcib);
-
-  gtpci->irq_res = bus_alloc_irq(pcib, 0, MIPS_HWINT0, RF_ACTIVE);
+  gtpci->irq_res = device_take_irq(pcib, 0, RF_ACTIVE);
   bus_intr_setup(pcib, gtpci->irq_res, gt_pci_intr, NULL, gtpci,
                  "GT64120 main irq");
 
+  pci_bus_enumerate(pcib);
+
+  /*
+   * Create child devices of ISA bus.
+   */
+  device_t *dev;
+
+  /* atkbdc keyboard device */
+  dev = device_add_child(pcib, 0);
+  dev->bus = DEV_BUS_PCI; /* XXX: ISA device workaround */
+  device_add_ioports(dev, 0, IO_KBD, IO_KBDSIZE);
+  device_add_irq(dev, 0, 1);
+
+  /* ns16550 uart device */
+  dev = device_add_child(pcib, 1);
+  dev->bus = DEV_BUS_PCI; /* XXX: ISA device workaround */
+  device_add_ioports(dev, 0, IO_COM1, IO_COMSIZE);
+  device_add_irq(dev, 0, 4);
+
+  /* rtc device */
+  dev = device_add_child(pcib, 2);
+  dev->bus = DEV_BUS_PCI; /* XXX: ISA device workaround */
+  device_add_ioports(dev, 0, IO_RTC, IO_RTCSIZE);
+  device_add_irq(dev, 0, 8);
+
+  /* i8254 timer device */
+  dev = device_add_child(pcib, 3);
+  dev->bus = DEV_BUS_PCI; /* XXX: ISA device workaround */
+  device_add_ioports(dev, 0, IO_TIMER1, IO_TMRSIZE);
+  device_add_irq(dev, 0, 0);
```

Comment:
Maybe it will be later parsed from a file, but at the moment it is a good idea to use mips_intr_t from mips/interrupt.h for `device_add_irq` like in sys/mips/rootdev.c  - `device_add_irq(dev, 0, MIPS_HWINT5);` instead of raw ints.

Reply:
These are ISA interrupts and are completely unrelated to MIPS interrupts. I think you remark will be addressed in #836.